File: flutter/lib/domain/usecases/get_expenses.dart
import 'package:smart_expense_manager/core/result.dart';
import 'package:smart_expense_manager/domain/models/expense.dart';
import 'package:smart_expense_manager/domain/repositories/expense_repository.dart';

class GetExpenses {
  final ExpenseRepository repository;

  GetExpenses(this.repository);

  Future<Result<List<Expense>>> call() {
    return repository.getExpenses();
  }

  Future<Result<List<Expense>>> callByDate(DateTime date) {
    return repository.getExpensesByDate(date);
  }

  Future<Result<List<Expense>>> callByCategory(Category category) {
    return repository.getExpensesByCategory(category);
  }
}

File: android_native/app/src/main/java/com/smartexpense/manager/domain/usecase/GetExpenses.kt
package com.smartexpense.manager.domain.usecase

import com.smartexpense.manager.domain.repository.ExpenseRepository
import javax.inject.Inject

class GetExpenses @Inject constructor(
    private val repository: ExpenseRepository
) {
    suspend operator fun invoke() = repository.getExpenses()
    
    suspend fun callByDate(date: java.util.Date) = repository.getExpensesByDate(date)
    
    suspend fun callByCategory(category: com.smartexpense.manager.domain.model.Category) = 
        repository.getExpensesByCategory(category)
}
